week0/homework/Person.js

```diff
@@ -0,0 +1,20 @@
+module.exports =  function (name,age)  {
+   let arr = [];
+   this.name = name;
+   this.age = age;
+   this.addPhone = function(phone) {
```

Comment:
This setup works (in conjunction with the map function alter) as long as the method is called only once. But if it's called more than once, instead of overwriting the existing phone number, you'll create an additional copy. (Try it out perhaps)

Also, since you're duplicating name and age data, it's not ideal and would also present problems. For example, what if you change the age property after calling addPhone? The last element of the array would have outdated data, and hence is not a good pattern to build on.